src/main/kotlin/org/rust/ide/inspections/RsTraitObjectInspection.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.isEdition2018
+import org.rust.lang.core.resolve.ref.deepResolve
+
+class RsTraitObjectInspection : RsLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor = object : RsVisitor() {
+        override fun visitTypeReference(typeReference: RsTypeReference) {
+            if (!typeReference.isEdition2018) return
+
+            if (typeReference.baseType?.path?.reference?.deepResolve() is RsTraitItem
+                || typeReference.traitType?.firstChild?.elementType == RsElementTypes.POLYBOUND
```

Comment:
Looks too hacky. Just add this
```
val RsTraitType.dyn: PsiElement? get() = node.findChildByType(RsElementTypes.DYN)?.psi
```
to `RsTraitType.kt`. Then use it here:

```suggestion
                || typeReference.traitType?.dyn == null
```

Also could you please ping your reviewer after resolving comments, b/c I don't receive any notifications on force-pushes to a PR, and it's hard to track them manually.